gprofiler/spark/spark_collector.py

```diff
@@ -0,0 +1,992 @@
+import contextlib
```

Comment:
This code was adapted from https://github.com/DataDog/integrations-core/blob/master/spark/datadog_checks/spark/spark.py - we'll need to add appropriate contribution. I will take care of it, leaving it here as a reminder.